sale_partner_operating_unit/models/sale_order.py

```diff
@@ -0,0 +1,35 @@
+# Copyright 2025 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.depends("partner_id.operating_unit_ids")
+    def _compute_team_id(self):
+        # For orders without a team, if the partner's ``operating_unit_ids``
+        # is set exactly one, we skip the computation of the team.
+        skip_teams_sales = self.filtered(
+            lambda order: not order.team_id
+            and order.partner_id.operating_unit_ids
+            and (len(order.partner_id.operating_unit_ids) == 1)
+        )
+        if skip_teams_sales:
+            self = self - skip_teams_sales
+        return super()._compute_team_id()
```

Comment:
nitpicking: 
```suggestion
        # For orders without a sales team, if the partner has exactly one operating unit,
        # we exclude those orders from team computation.
        single_operating_unit_orders_without_team = self.filtered(
           lambda so: not so.team_id
           and so.partner_id.operating_unit_ids
           and len(so.partner_id.operating_unit_ids) == 1
       )
       if single_operating_unit_orders_without_team:
           self = self - single_operating_unit_orders_without_team
        return super()._compute_team_id()
```

alternative variable name: `unassined_single_operating_unit_orders`

Reply:
Thanks @vvrossem @SilvioC2C